pex/resolve/lockfile/updater.py

```diff
@@ -110,9 +232,13 @@ def create(
     pip_configuration = attr.ib()  # type: PipConfiguration
 
     @contextmanager
-    def _calculate_requirement_configuration(self, locked_resolve):
-        # type: (LockedResolve) -> Iterator[RequirementConfiguration]
-        if not self.update_constraints_by_project_name:
+    def _calculate_requirement_configuration(
+        self,
+        locked_resolve,  # type: LockedResolve
+        pin_all=False,  # type: bool
```

Comment:
Given that this is `pin_all` (implying that a different scope of pinning might be necessary/useful), should the `--pin` flag maybe start as either an enum, or perhaps a list option with an initial special-case value of e.g. `:all:`?

Reply:
I don't think so. The two existing modes of lock update are:
+ no args: Update the lock given its original input root requirements (these are stored in a lockfile)
+ `-p one -p "two<2" -p three==1.0` ...: Update the lock pinning everything else and just updating (or adding) these projects if possible.

So this just offers the only missing mode as far as I can tell - update the lock keeping everything `--pin`ned.